### DISCUSSION TITLE
Are Next JS deployments in Vercel zero downtime?

### DISCUSSION THREAD

**Discussion by jaeyow** [vercel/vercel#6518] [General] [answered]
I would like to ask if NextJS deployments in Vercel are zero-downtime? I am in the process of exploration to figure out if Next JS is feasible for my next project. 

In my typical Production web applications, I would have a couple of EC2 instances, and run a rolling deployment so that there is zero downtime while the deployment is in progress. 

I have tried the starter Next JS project, and have deployed them in Vercel, yes they feel like they are zero downtime, however  is this the case though? Would larger Next JS projects with longer deployment cycles still have the same zero downtime experience? 

If not, what is the recommendation here?

Is it possible to run 2 Next JS deployments under a load balancer to achieve this? Or am I talking nonsense, as there is no need for this at all?

**Comment by leerob** [comment, marked as answer]
I'm following entirely following - are you asking about a migration path to Vercel (and if it's zero-downtime, including domains and DNS) or once you're already on Vercel and doing another deployment?

Related: https://vercel.com/support/articles/zero-downtime-migration-for-dns

**Reply by jaeyow** [reply]
yeah, once I'm already in Vercel, and have just pushed my code changes to git, when a deployment is triggered and in progress, will my visitors experience a downtime at all when the site switches from old to new?

Thanks @leerob for the links about DNS and domains, handy to know that.

**Reply by leerob** [reply]
Got it! No, there will not be downtime 😁 Vercel will only replace your live production website if the new build completes successfully. Then, traffic will be pointed to the new deployment with no downtime.

**Reply by jaeyow** [reply]
This is awesome @leerob . I'm coming from the traditional two EC2 instances and rolling deployments to avoid downtime during deployment camp.

So to see how Vercel approaches these Next JS deployments, makes the deployment/infrastructure story simple, it looks like I won't even need to put up a load balancer before the website anymore!

**Reply by leerob** [reply]
Glad I could help 😄 No load balancer needed!
